tests/hf/test_tapering.py

```diff
@@ -0,0 +1,148 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for functions needed for qubit tapering.
+"""
+import pennylane as qml
+import pytest
+from pennylane import numpy as np
+from pennylane.hf.tapering import clifford, _observable_mult, _simplify, transform_hamiltonian
+
+
+@pytest.mark.parametrize(
+    ("obs_a", "obs_b", "result"),
+    [
+        (
+            qml.Hamiltonian(np.array([-1.0]), [qml.PauliX(0) @ qml.PauliY(1) @ qml.PauliX(2)]),
+            qml.Hamiltonian(np.array([-1.0]), [qml.PauliX(0) @ qml.PauliY(1) @ qml.PauliX(2)]),
+            qml.Hamiltonian(np.array([1.0]), [qml.Identity(0)]),
+        ),
+        (
+            qml.Hamiltonian(
+                np.array([0.5, 0.5]), [qml.PauliX(0) @ qml.PauliY(1), qml.PauliX(0) @ qml.PauliZ(1)]
+            ),
+            qml.Hamiltonian(
+                np.array([0.5, 0.5]), [qml.PauliX(0) @ qml.PauliX(1), qml.PauliZ(0) @ qml.PauliZ(1)]
+            ),
+            qml.Hamiltonian(
+                np.array([-0.25j, 0.25j, -0.25j, 0.25]),
+                [qml.PauliY(0), qml.PauliY(1), qml.PauliZ(1), qml.PauliY(0) @ qml.PauliX(1)],
+            ),
+        ),
+    ],
+)
+def test_observable_mult(obs_a, obs_b, result):
+    r"""Test that observable_mult returns the correct result."""
+    o = _observable_mult(obs_a, obs_b)
+    assert o.compare(result)
+
+
+@pytest.mark.parametrize(
+    ("hamiltonian", "result"),
+    [
+        (
+            qml.Hamiltonian(
```

Comment:
I fear that this test is a bit too simple for a very general function like simplify. Can we think of difficult cases where it may fail? For example, what happens if you add operators with equal magnitude but different sign? They should add to zero, but what do we get? Other potentially problematic cases we could think of?

Reply:
Added few more tests including the one in the comment.